models/model_interface.py

```diff
@@ -1,21 +1,195 @@
+    ) -> np.ndarray:
+        """Predicts the probability of a positive outcome for each example in
+        the dataset.
+
+        Args:
+            dataset: A dataset without the label column and pre-transformed
+                using the class's RecourseAdapter.
+
+        Returns:
+            A 1-dimensional numpy array of positive class probabilities."""
+
+    @abc.abstractmethod
+    def _predict(
+        self, dataset: recourse_adapter.EmbeddedDataFrame
+    ) -> np.ndarray:
+        """Predicts the class label for each example in the dataset.
+
+        The class predictions should be in embedded space.
+
+        Args:
+            dataset: A dataset without the label column and pre-transformed
+                using the class's RecourseAdapter.
+
+        Returns:
+            A 1-dimensional numpy array of class predictions."""
+
+    @abc.abstractmethod
+    def to_dice_model(self) -> dice_ml.Model:
+        """Returns a dice_ml Model based on this model."""
+
+    def predict_pos_proba(self, dataset: pd.DataFrame) -> pd.Series:
+        """Predicts the probability of a positive outcome for each example in
+        the dataset.
+
+        If the dataset includes the label column, the column is automatically
+        dropped. The dataset should be passed in its unprocessed human-readable
+        format.
+
+        Args:
+            dataset: The dataset to predict over.
+
+        Returns:
+            A series where element i of the series is the probability of a
+            positive outcome for element i of the dataset."""
+        dataset = self.adapter.transform(dataset)
+        if self.adapter.label_name in dataset.columns:
+            dataset = dataset.drop(self.adapter.label_name, axis=1)
+        proba = self._predict_pos_proba(dataset)
+        return pd.Series(proba, index=dataset.index)
+
+    def predict(self, dataset: pd.DataFrame) -> pd.Series:
+        """Predicts the class for each example in the dataset.
+
+        If the dataset includes the label column, the column is automatically
+        dropped. The dataset should be passed in its unprocessed human-readable
+        format.
+
+        Args:
+            dataset: The dataset to predict over.
+
+        Returns:
+            A series where element i of the series is the predicted
+            human-readable class of example i of the dataset."""
+        dataset = self.adapter.transform(dataset)
+        if self.adapter.label_name in dataset.columns:
+            dataset = dataset.drop(self.adapter.label_name, axis=1)
+        y_pred = self._predict(dataset)
+        y_pred_series = pd.Series(y_pred, index=dataset.index)
+        return self.adapter.inverse_transform_label(y_pred_series)
+
+    def predict_pos_proba_series(self, x: pd.Series) -> float:
+        """Predicts the probability of a positive outcome for a single data
+        example.
+
+        If the example includes a label, the label is automatically dropped.
```

Comment:
I don't see the label dropping taking place in this function. Can you point out what I'm missing? Same goes for the predict_series() function on line 121.

Reply:
Both *_series functions rely on other functions which drop the label. For example, predict_series() uses predict(), and predict drops the class label if present. I added a comment in the file to explain this.